pandas/core/reshape/merge.py

```diff
@@ -952,7 +970,7 @@ def _get_join_info(self):
                     )
                 else:
                     join_index = self.right.index.take(right_indexer)
-                    left_indexer = np.array([-1] * len(join_index))
+                    left_indexer = np.array([-1] * len(join_index), dtype=np.intp)
```

Comment:
should be using the `np.repeat` syntax similar to above?

Reply:
either way works.  towards the end of this process i might try to standardize them, but thats pretty low priority